os_traits/const.py

```diff
@@ -67,9 +67,55 @@
 # ref: https://en.wikipedia.org/wiki/AMD-V
 HW_CPU_X86_SVM = _CPU_X86_NS + 'svm'
 
+# All hypervisor specific features are prefixed with this namespace
+_VIRT_NS = 'virt:'
+
+# All libvirt specific features are prefixed with this namespace
+_VIRT_LIBVIRT_NS = _VIRT_NS + 'libvirt:'
+
+# All Hyper-V specific features are prefixed with this namespace
+_VIRT_HYPERV_NS = _VIRT_NS + 'hyperv:'
+
+_VIRT_HYPERV_VM_GENERATION = _VIRT_HYPERV_NS + 'vm_generation:'
+VIRT_HYPERV_VM_GENERATION_1 = -_VIRT_HYPERV_VM_GENERATION + '1'
+VIRT_HYPERV_VM_GENERATION_2 = -_VIRT_HYPERV_VM_GENERATION + '2'
+
+VIRT_HYPERV_SECURE_BOOT = _VIRT_HYPERV_NS + 'secure_boot'
+VIRT_HYPERV_SHIELDED_VMS =_VIRT_HYPERV_NS + 'shielded_vms'
+
+# RemoteFX suffix
+_VIRT_HYPERV_REMOTEFX = _VIRT_HYPERV_NS + 'remotefx:'
+
+# VRAM is requested in MB
+_VIRT_HYPERV_REMOTEFX_MEMORY = _VIRT_HYPERV_REMOTEFX + 'vram:'
+VIRT_HYPERV_REMOTEFX_MEMORY_64 = _VIRT_HYPERV_REMOTEFX_MEMORY + '64'
+VIRT_HYPERV_REMOTEFX_MEMORY_128 = _VIRT_HYPERV_REMOTEFX_MEMORY + '128'
+VIRT_HYPERV_REMOTEFX_MEMORY_256 = _VIRT_HYPERV_REMOTEFX_MEMORY + '256'
+VIRT_HYPERV_REMOTEFX_MEMORY_512 = _VIRT_HYPERV_REMOTEFX_MEMORY + '512'
+VIRT_HYPERV_REMOTEFX_MEMORY_1024 = _VIRT_HYPERV_REMOTEFX_MEMORY + '1024'
```

Comment:
While I recognize the value of the functionality provided by the concept of `power_stepping` I worry _alot_ about the way in which the relatively simple original data model of resource providers with classes of inventories is growing limbs. My inclination would be that rather than extending the model to encompass lots of use cases which instead focus on the core use cases and address the edge cases at the edge (as in, in other tools).

I certainly appreciate that `power_stepping` is a generic solution to the problem presented by the OP, and that's great because it can solve more than that problem, but it is...smelly?

I feel a bit like the need to "quickly reject a request"  need to be evaluated carefully and with due consideration for simplicity.